File: backend/src/main/java/com/example/masters/dto/rule/RuleRequest.java
package com.example.masters.dto.rule;

import lombok.Data;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
public class RuleRequest {
    private String ruleType;        // "SENSOR" або "TIME"

    private UUID deviceId;        // сенсор
    private String operator;        // ">", "<", "="
    private Double threshold;

    private UUID actionDeviceId;    // пристрій для керування
    private String actionType;      // "on", "off"
    private Double actionValue;       // 24

    private LocalDateTime triggerTime;  // для TIME правил
}

File: frontend/app/src/main/java/com/example/smartlab/model/RuleRequest.kt
package com.example.smartlab.model

import java.time.LocalDateTime

data class RuleRequest(
    val ruleType: String,          // "SENSOR" або "TIME"
    val deviceId: String?,           // сенсор
    val operator: String?,         // ">", "<", "="
    val threshold: Double?,        // порогове значення
    val actionDeviceId: String?,     // пристрій для керування
    val actionType: String?,       // "on", "off"
    val actionValue: Double?,      // 24
    val triggerTime: String?, // для TIME правил
    val active: Boolean,
    val mirrorRule: Boolean
)